lf-edge/edge-home-orchestration-go
# Moved the MNEDC server/client interrupt logic to sigmgr

# Description

Signal interrupt logic for closing MNEDC server/client has been moved from the mnedcmgr package to sigmgr.
Also, Unit tests have been updated for the above change.

Fixes #172 

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?
1. Run MNEDC server in Main NAT
```
$ ./build.sh mnedcserver
```
2. Run MNEDC client in Sub NAT
```
$ ./build.sh mnedcclient
```
3. After they join, stop their docker containers in both devices by running ```$ docker stop edge-orchestration```
4. Observe following in the MNEDC server:
```
2020/11/13 08:03:33 sigmgr.go:39: [sigmgr] Received Signal: terminated
2020/11/13 08:03:33 sigmgr.go:44: [sigmgr] [MNEDC Server] Server Closed
2020/11/13 08:03:33 sigmgr.go:48: [sigmgr] [MNEDC Client] Client not alive
```
5. Observe following in the MNEDC client:
```
2020/11/13 08:03:09 sigmgr.go:39: [sigmgr] Received Signal: terminated
2020/11/13 08:03:09 sigmgr.go:42: [sigmgr] [MNEDC Server] Server not alive
2020/11/13 08:03:09 sigmgr.go:50: [sigmgr] [MNEDC Client] Client Closed
```

**Test Configuration**:
* Firmware version: Ubuntu 18.04
* Hardware: x86-64

# Checklist:
